docs/design/jet/023-kafka-connect-source-connectors.md

```diff
@@ -0,0 +1,340 @@
+[common](https://docs.confluent.io/platform/current/installation/configuration/connect/source-connect-configs.html#kconnect-long-source-configuration-properties-for-cp)
+while the other connector-specific. Refer to the individual connector documentation for connector-specific configuration
+properties.
+
+Only `name` and `connector.class` properties are mandatory.
+
+Example pipeline using Kafka-based `RandomSourceConnector` to generate random numbers
+
+```java
+        Properties randomProperties = new Properties();
+        //mandatory properties
+        randomProperties.setProperty("name", "random-source-connector");
+        randomProperties.setProperty("connector.class", "sasakitoa.kafka.connect.random.RandomSourceConnector");
+        
+        //common properties
+        randomProperties.setProperty("tasks.max", "1");
+        
+        //connector-specific properties
+        randomProperties.setProperty("generator.class", "sasakitoa.kafka.connect.random.generator.RandomInt");
+        randomProperties.setProperty("messages.per.second", "1000");
+        randomProperties.setProperty("topic", "test");
+        randomProperties.setProperty("task.summary.enable", "true");
+
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(KafkaConnectSources.connect(randomProperties))
+                .withoutTimestamps()
+                .map(record -> Values.convertToString(record.valueSchema(), record.value()))
+                .writeTo(AssertionSinks.assertCollectedEventually(60,
+                        list -> assertEquals(ITEM_COUNT, list.size())));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.addJar(Objects.requireNonNull(this.getClass()
+                .getClassLoader()
+                .getResource("random-connector-1.0-SNAPSHOT.jar"))
+        .getPath()
+        );
+
+        Job job = createHazelcastInstance().getJet().newJob(pipeline, jobConfig);
+```
+
+#### Client Related Changes
+
+No changes required
+
+### Technical Design
+
+In the `hazelcast-jet-contrib` repository we already have an initial implementation of the `KafkaConnectSources`.
+
+`KafkaConnectSources` instantiates the `KafkaConnectSource`, which is used then to create as task object which performs
+the actual ingestion logic.
+Everything is orchestrated by the `SourceBuilder`
+methods: `createFn`, `fillBufferFn`, `createSnapshotFn`, `restoreSnapshotFn`, `destroyFn`.
+
+The current implementation uses `SourceBuilder` to create a stream source of `SourceRecord`.
+`SourceRecord` is Kafka data type returned by the Kafka Source Connectors for further processing.
+
+The plan is to move the code to the main hazelcast repository and make it production ready with the following steps:
+
+- Move Kafka Connect to Hazelcast Extensions module
+- Add Kafka connect source scaling (currently Hazelcast Platform will instantiate a single task for the specified source
```

Comment:
Personally I'd add snapshotting support as a separate point. Not as easy as I thought ;)

Reply:
already there: https://github.com/hazelcast/hazelcast/pull/23303/files/b002cf0da1f0357cddf78b4e5b20f3ade8e555d1#diff-4e57279648cfda96eda7b4645a2441607d164df5c44b8fc89f89cff1a44fd844R195-R199